Classes/Controller/HaikuController.php

```diff
@@ -0,0 +1,126 @@
+        switch ($action) {
+            case 'show':
+                return $this->showAction((int)$_REQUEST['tx_examples_haiku']['haiku'] ?? 0);
+            case 'findByTitle':
+                return $this->findByTitleAction($_REQUEST['tx_examples_haiku']['haiku_title'] ?? '');
+            case 'list':
+                return $this->listAction();
+            default:
+                return $this->errorAction();
+        }
+    }
+
+    private function initView(): void
+    {
+        $this->view = GeneralUtility::makeInstance(StandaloneView::class);
+        $this->view->setLayoutRootPaths([
+            GeneralUtility::getFileAbsFileName('EXT:examples/Resources/Private/Layouts'),
+        ]);
+        $this->view->setPartialRootPaths([
+            GeneralUtility::getFileAbsFileName('EXT:examples/Resources/Private/Partials'),
+        ]);
+        $this->view->setTemplateRootPaths([
+            GeneralUtility::getFileAbsFileName('EXT:examples/Resources/Private/Templates'),
+        ]);
+        $this->view->setFormat('html');
+        $this->view->setTemplate('Haiku/Default');
+        $this->view->assignMultiple([
+            'settings' => $this->conf['settings'] ?? [],
+        ]);
+    }
+
+    private function showAction(int $haiku): string
+    {
+        $this->view->assignMultiple([
+            'haiku' => $this->haikuRepository->findByUid($haiku),
+        ]);
+        $this->view->setTemplate('Haiku/Show');
+        return $this->view->render();
+    }
+
+    private function listAction(): string
+    {
+        $this->view->assignMultiple([
+            'haikus' => $this->haikuRepository->findAll(),
+        ]);
+        $this->view->setTemplate('Haiku/List');
+        return $this->view->render();
+    }
+
+    private function errorAction(): string
+    {
+        return $this->listAction();
+    }
+
+    private function findByTitleAction(string $haikuTitle): string
+    {
+        $this->view->assignMultiple([
+            'haiku' => $this->haikuRepository->findByTitle($haikuTitle),
+        ]);
+        $this->view->setTemplate('Haiku/Show');
+        return $this->view->render();
+    }
+
+    private function loadFlexFormSettings(): void
+    {
+        if (!is_string($this->cObj->data['pi_flexform'])) {
+            return;
+        }
+        $flexFormData = GeneralUtility::makeInstance(FlexFormService::class)
+            ->convertFlexFormContentToArray($this->cObj->data['pi_flexform']);
+        $this->cObj->data['pi_flexform'] = $flexFormData;
+        if (!isset($this->conf['settings']) or !is_array($this->conf['settings'])) {
+            $this->conf['settings'] = [];
+        }
+        if (isset($flexFormData['settings']) && is_array($flexFormData['settings'])) {
+            foreach ($flexFormData['settings'] as $key => $flexFormValue) {
+                if ($flexFormValue) {
+                    $this->conf['settings'][$key] = (string)$flexFormValue;
```

Comment:
Can $flexFormValue be something else than a string? It is from an XML attribute, so is should be always a string, or am I wrong?

Reply:
I think it could be an integer or some such?

Reply:
Hmm, you may be right, when the TCA type is a number and the FlexFormService converts it accordingly. Don't know really, I'm avoiding flexform whenever possible. But should we check.